activerecord/test/models/sharded/blog_post_destroy_async.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Sharded
+  class BlogPostDestroyAsync < ActiveRecord::Base
```

Comment:
I was thinking we should try inheriting from `BlogPost` to avoid setting the table name, query constraints and associations that do not change.
Unless it will start causing conflicts

Reply:
This made sense, but I more or less copied the `EssayDestroyAsync` and `BookDestroyAsync` examples (they didn't use inheritance). It also makes it easier to debug an isolated test because it requires less indirection.

There's definitely a tradeoff, we can address this when I PR it to `rails`.